### ISSUE TITLE
Invalid JSON expression when using Operator "*"

### ISSUE THREAD

**Issue by darkklin** [json-path/JsonPath#558]
JSON exmple 
 `{
    "request": {
        "Target": "Offer",
        "Format": "json",
        "Service": "HasOffers",
        "Version": "2",
        "NetworkToken": "",
        "Method": "findAll",
        "fields": [
            "id",
            "preview_url"
        ],
        "filters": {
            "status": "active"
        }
    },
    "response": {
        "status": 1,
        "httpStatus": 200,
        "data": {
            "87": {
                "Offer": {
                    "id": "87",
                    "preview_url": "http://TEST.COM"
                }
            },
            "42": {
                "Offer": {
                    "id": "42",
                    "preview_url": "http://TEST.COM"
                }
            },
            "41": {
                "Offer": {
                    "id": "41",
                    "preview_url": "http://TEST.COM"
                }
            }
        }
  }
}`
**Goal** 
pull/print the ids number from the json

**Code** 
`Response response = given().get(
				"")			.then().assertThat().statusCode(200).and().contentType(ContentType.JSON).extract().response();`
`JsonPath activeOffersJson = response.jsonPath();`
CODE fail here -> `System.out.println(activeOffersJson.get("response.data.[*].Offer.id").toString());`

**Error**
`java.lang.IllegalArgumentException: Invalid JSON expression:
Script1.groovy: 1: unexpected token: [ @ line 1, column 41.
   dJsonRootObject.response.data.[*].Offer.
                                                       ^`

**Screen shot of my code and console** 
![Screenshot_919](https://user-images.githubusercontent.com/25209561/61857662-3efff000-aecd-11e9-828e-16bbab8da3af.png)

The  path ("response.data.[*].Offer.id") works fine when I test it here 
http://jsonpath.com/?fbclid=IwAR1CwcB5q3qB8-_HeYBVzOf-xQLavqs5LHOi9V5rCsBzTX8kxVTsUf7fhI0

**Comment by hf-kklein** [comment]
Personally, I find it hard to follow your description. Could you please post a minimal but fully working example **as code** that demonstrates the error?

**Comment by darkklin** [comment]
I edited the post, Hope this is more understandable

**Comment by darkklin** [comment]
Oh, sh*it  used the wrong library "import io.restassured.path.json.JsonPath;"  
LOL
